docs: [vision-on-sample-app][6/n] Common Views

Ahmed-CIO opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #617
* #613
* #612
* #611
* #610
* __->__ #609
* #608
* #606
* #605
* #604

## Context

This is a PR in a series of PRs to build sample app that runs Identify and Track in VisionPro
The app is built to be an interactive walk through guidance for developers who are integrating Customer.io Swift SDK in their VisioPro apps.
For more context:

- See the project [one pager](https://www.notion.so/custio/Spacial-Identify-and-Tracking-6587d848bbde494095c0585237326ed3?pvs=4)
- See [Linear Issue](https://linear.app/customerio/issue/MBL-115/sample-app-for-visionpro)


## PR Summary
This PR adds punch of generic SwiftUI view
- `CommonLayoutView` this view will be used to wrap all content and handle bunch of the navigation logic. Including displaying the `Logo`, the `TutroialsMenuView`, and error messages
- `LineView` custom view for flexible divider
- `FloatingTitleTextView` custom text field to ensure the field placeholder becomes labels when the field is filled.


## Test plan
- Open `VisionOs.xcproject`
- Hit CMD+R to run the project and make sure you have VisionOS simulator installed and selected

levibostian commented on Apps/VisionOS/VisionOS/CommonViews/CommonLayoutView.swift:
```diff
@@ -0,0 +1,65 @@
+    func makeBody(configuration: Configuration) -> some View {
+        Button(configuration)
+            .buttonBorderShape(.roundedRectangle(radius: 10))
+            .background(Color(rgba: 0x333333FF))
```
Will this color be used in multiple parts of the app? If so, I think it's best to not have it hard-coded in a View. 

Perhaps use extensions so we could do: `Color.foo` to reference it?

Ahmed-CIO replied:
So far this is only used here. Agreed if at any point I needed to repeat it, it will be extracted out

levibostian commented on Apps/VisionOS/VisionOS/CommonViews/TutorialsMenuView.swift:
```diff
@@ -0,0 +1,45 @@
+import SwiftUI
+
+@ViewBuilder func button(
+    forLink link: InlineNavigationLink,
+    state: AppState
+) -> some View {
+    Button {
+        state.navigationPath = [link]
+    } label: {
+        Text(link.menuTitle)
+        if state.visitedLinks.contains(link) {
+            Image(systemName: "checkmark.circle.fill")
+        }
+    }
+}
+
+func getEnabledNavigationLinks(_ state: AppState) -> [InlineNavigationLink] {
+    if state.workspaceSettings.isSet(), state.profile.loggedIn {
+        return InlineNavigationLink.allCases
+    }
+
+    if state.workspaceSettings.isSet() {
+        return InlineNavigationLink.preIdentifyLinks
+    }
+
+    return InlineNavigationLink.preInitializationLinks
+}
```
This code seems un-related to `TutorialsMenuView`. If that's true, could we move this code into another file?

Ahmed-CIO replied:
It belongs here. The button view builder decides whether to show a checkmark in the menu entry based on whether the said link is visited. The `getEnabledNavigationLinks` we can argue whether it belongs here or in the shared view model. Since it is only used here, it seems appropriate to keep it here.

levibostian commented on Apps/VisionOS/VisionOS/ViewUtils/InlineNavigationLink.swift:
```diff
@@ -22,4 +22,49 @@ enum InlineNavigationLink: String, Codable, CaseIterable {
         self = link
     }
+
+    var menuTitle: String {
+        // TODO: menu title for each screen
+        switch self {
+        case .sampleAppIntro:
+            ""
+        case .install:
+            ""
+        case .setup:
+            ""
+        case .customerIOIntro:
+            ""
+        case .identify:
+            ""
+        case .howToTestIdentify:
+            ""
+        case .profileAttributes:
+            ""
+        case .howToTestProfileAttributes:
+            ""
+        case .deviceAttributes:
+            ""
+        case .howToTestDeviceAttributes:
+            ""
+        case .track:
+            ""
+        case .howToTestTrack:
+            ""
+        }
+    }
```
Non-blocking question: Is it possible for each screen's View to decide what the menu title is? I am OK with this if that's preferred.

Ahmed-CIO replied:
You are spot on. Each of these cases will get the title from the relevant screen. Hence it is currently blank "to indicate the missing title". The only reason this is added here as an associated property, is to avoid the need for each callsite to do the logic of this switch case.

levibostian reviewed (approved):
This may not be in scope for this app, but I still want to ask. 

Could some of these generic SwiftUI views be re-used with the SwiftUI iOS sample app?

Ahmed-CIO commented:
Apologies for that, gh-stack pushes seems to cause too much noise due to the fact that it pushes the entire stack for every push. These changes will carry on #622 

Meanwhile, I addressed the comments in this PR
